src/Sylius/Bundle/CoreBundle/PaymentRequest/Checker/PaymentRequestIntegrityChecker.php

```diff
@@ -29,16 +30,19 @@ public function __construct(
     public function check(PaymentRequestHashAwareInterface $command): PaymentRequestInterface
     {
         $hash = $command->getHash();
-        Assert::notNull($hash, 'Payment request hash cannot be null.');
 
         $paymentRequest = $this->paymentRequestRepository->findOneByHash($hash);
-        Assert::notNull($paymentRequest, sprintf('Payment request (hash "%s") not found.', $hash));
+        if (null === $paymentRequest) {
+            throw new PaymentRequestNotFoundException();
+        }
 
         /** @var PaymentMethodInterface $paymentMethod */
         $paymentMethod = $paymentRequest->getMethod();
 
         $gatewayConfig = $paymentMethod->getGatewayConfig();
-        Assert::notNull($gatewayConfig, 'Gateway config cannot be null.');
+        if (null === $gatewayConfig) {
+            throw new NullPayumTokenException();
+        }
```

Comment:
It doesn't sound logical for me. If there's no gateway config then we throw the payum token is null. 

1) We shouldn't reveal we're using Payum under the hood.
2) There should be a correlation between the exception and check. Maybe something like `NoGatewayConfigException`?